client/src/modules/vouchers/vouchers.service.js

```diff
@@ -15,18 +15,78 @@ VoucherService.$inject = [
  * includes some utilities that are useful for voucher pages.
  */
 function VoucherService(Api, $http, util, TransactionTypeStore, Modal,
-  Filters) {
+  Filters, Periods, Languages, $httpParamSerializer) {
   var service = new Api('/vouchers/');
+  var voucherFilters = new Filters();
+  var filterCache = new AppCache('voucher-filters');
 
   // @todo - remove this reference to baseUrl
   var baseUrl = '/journal/';
-  var filter = new Filters();
 
   service.create = create;
   service.reverse = reverse;
   service.transactionType = transactionType;
   service.openSearchModal = openSearchModal;
-  service.formatFilterParameters = formatFilterParameters;
+
+  // service.formatFilterParameters = formatFilterParameters;
+  service.filters = voucherFilters;
+  service.cacheFilters = cacheFilters;
+  service.removeFilter = removeFilter;
+  service.loadCachedFilters = loadCachedFilters;
+  service.download = download;
+
+  voucherFilters.registerDefaultFilters([
+    { key : 'period', label : 'TABLE.COLUMNS.PERIOD', valueFilter : 'translate' },
+    { key : 'custom_period_start', label : 'PERIODS.START', valueFilter : 'date' },
```

Comment:
Custom period start and end should now have a comparison symbol to show that the dates are before and after the individual dates. 

They can be added like { comparitor : '<' }

(As of https://github.com/IMA-WorldHealth/bhima-2.X/pull/1774)